### PR TITLE
Close ui-dashboard React Doctor backlog

### PR DESCRIPTION
**Pull request by chapati23**
## Summary

- Bumps `react-doctor` to 0.1.6 and wires the React Doctor ESLint plugin plus the companion effect plugin into the dashboard lint config.
- Adds a UI CI score floor requiring `react-doctor --full --score --offline` to return `100`.
- Removes the targeted React Doctor debt in touched pool-detail, oracle-tab, breach-history, and hook-test surfaces, then closes the related `BACKLOG.md` items.

## What This PR Changes

- Splits pool detail rendering into smaller helper components and moves legacy pool-id canonicalization to the server route.
- Extracts oracle snapshot table/row rendering from `OracleTab` while keeping its bounded search and dedicated chart query behavior intact.
- Replaces the breach history panel's cluster of local state setters with a reducer without changing its search/sort/pagination query contract.
- Documents the remaining intentional React Doctor silences, including the `HASURA_TIMEOUT_MS` compatibility re-export.

## Invariants

- `ui-dashboard` full React Doctor score is exactly `100` with zero active diagnostics.
- Oracle and breach-history tables keep deterministic server-side ordering, visible count/search degradation, and existing pagination reset behavior.
- Pool TVL/volume/reserve charts remain independent from tab table pagination and preserve upstream threshold loading/error handling.
- Server code continues importing shared Hasura timeout constants from `@/lib/hasura-timeout`, not the client-only GraphQL hook module.

## Degraded Behavior

- Oracle count-query failure still preserves the last known total and renders a visible warning.
- Bounded oracle search still discloses when it is capped or falls back to the bootstrap window.
- Pool detail missing/error/loading states continue rendering explicit error or skeleton states.
- Production build still emits the pre-existing `metadataBase` warning; no new build failure is introduced.

## Intentional Non-goals

- No schema, indexer writer, or GraphQL shape changes.
- No broad design-system migration for React Doctor design-opinion rules.
- No migration to ES2023 array helpers until the dashboard runtime target changes.

## Test plan

- `./tools/trunk fmt`
- `./tools/trunk check`
- `pnpm --filter @mento-protocol/ui-dashboard lint`
- `pnpm --filter @mento-protocol/ui-dashboard typecheck`
- `pnpm --filter @mento-protocol/ui-dashboard react-doctor --full --score --offline` => `100`
- `pnpm --filter @mento-protocol/ui-dashboard test` (`116` files / `1998` tests)
- `pnpm dashboard:build`
- Pre-push hook also ran repo checks, indexer codegen, dashboard coverage, and metrics-bridge tests successfully.

---

> [!NOTE]
> **Medium Risk**
> Medium risk due to changes in the pool detail route’s redirect/canonicalization behavior and refactors of tab/panel rendering that could affect navigation and data-fetch gating; CI/lint dependency bumps may also introduce new tooling failures.
> 
> **Overview**
> **Closes the ui-dashboard React Doctor backlog and hardens the gate.** CI now enforces both a PR diff scan and a **full `react-doctor` score floor of 100**.
> 
> Updates dashboard linting by bumping `react-doctor` to `0.1.6`, re-enabling the `react-doctor` ESLint plugin (with mirrored suppressions to keep `pnpm lint` quiet), and adding `eslint-plugin-react-you-might-not-need-an-effect` with targeted rule disables.
> 
> Refactors several previously-suppressed UI surfaces to remove diagnostics: pool detail page canonicalization redirects move server-side in `pool/[poolId]/page.tsx`, `PoolDetailPageClient` is split into smaller helpers and avoids rendering tab chrome when the pool is missing, `OracleTab` extracts table/row rendering components, and `BreachHistoryPanel` consolidates related UI state into a reducer. Tests are adjusted/added to cover missing-pool behavior and to remove hook-shaped mock naming that required React Hooks rule overrides.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 01ef2a9538371893f392c882df1ea87cf44ad900. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by chapati23** [comment]
@claude review

**Comment by chapati23** [comment]
@claude addressed the review notes in `3150a02a22b1e752fa64d6b6f9c01201ce91a6e0`:

- CI score floor now strips whitespace from the captured React Doctor score before comparing to `100`.
- `OracleTab` uses a module-level `SEARCH_ORDER_BY` constant for the bounded-search ordering, and the file now imports React hooks directly for consistency.
- Pool detail tab visibility is memoized so the tab list identity stays stable unless the visibility inputs change.
- `BreachHistoryPanelInner` now uses a named props type instead of a large inline annotation.
- `ui-dashboard/AGENTS.md` now documents the split between high-noise React Doctor ESLint rules and the standalone full-score/diff gate.

I re-ran the local verification before pushing: `./tools/trunk fmt`, `./tools/trunk check`, dashboard lint/typecheck, React Doctor full score (`100`), dashboard tests (`116` files / `1998` tests), and `pnpm dashboard:build`. The pre-push hook also passed.

File: ui-dashboard/src/app/pool/[poolId]/_components/pool-detail-page-client.tsx [hunk 1]
```diff
@@ -391,86 +277,290 @@ function PoolDetail() {
...
+        normalizedPoolId={normalizedPoolId}
+        limit={limit}
+        pool={pool}
+        activeSearch={activeSearch}
+        setTabSearch={setTabSearch}
+        tradingLimits={tradingLimits}
+        tradingLimitsError={tradingLimitsError}
+        fpmmPool={fpmmPool}
+        network={network}
+      />
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/pool/[poolId]/_components/pool-detail-page-client.tsx]: Fixed in `f8d410e`: the tablist and tab panel are now suppressed once the pool detail query resolves to a missing pool, while keeping hook order stable and preserving the visible not-found error. I also added a regression test that asserts the missing-pool state renders no `tablist`/`tabpanel` and does not invoke the LP tab query.

Verified locally with focused pool-page test, dashboard lint/typecheck, React Doctor score `100`, `./tools/trunk check`, and the pre-push hook.

File: ui-dashboard/src/app/pool/[poolId]/page.tsx [hunk 2]
```diff
@@ -40,6 +55,24 @@ function buildDescription(data: PoolOgData): string {
...
+      params.set(key, value);
+    }
+  }
+  return params;
+}
+
+function routeCanonicalPoolId(poolId: string): string {
+  const chainId =
+    extractChainIdFromPoolId(poolId) ?? NETWORKS[DEFAULT_NETWORK].chainId;
+  return normalizePoolIdForChain(poolId, chainId);
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/pool/[poolId]/page.tsx:73]: Fixed in follow-up PR #385 (head `9f14460`). Raw `/pool/0x...` routes no longer default to Celo; they only canonicalize when a configured explicit `chainId` is present, strip that query param on redirect, and otherwise go to `/pools`. I also added server-route tests for raw+chain, bare raw, unsupported namespaced chains, supported namespaced routes, and leading-zero canonicalization.

File: ui-dashboard/package.json [hunk 3]
```diff
@@ -56,10 +56,11 @@
     "axe-core": "^4.11.4",
     "eslint": "^10.0.3",
     "eslint-plugin-jsx-a11y": "^6.10.2",
+    "eslint-plugin-react-you-might-not-need-an-effect": "^0.10.1",
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/package.json:59]: Addressed in follow-up PR #385 (head `9f14460`). The manifest and lockfile now exact-pin the plugin specifier to `0.10.1`; I also verified the npm tarball directly with `curl -I` returning HTTP 200 and `pnpm install --frozen-lockfile` passing locally before pushing.
